textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -93,13 +109,13 @@ public ITuple getNextTuple() throws DataFlowException {
             int schemaIndex = sourceTuple.getSchema().getIndex(SchemaConstants.SPAN_LIST_ATTRIBUTE.getFieldName());
             List<Span> spanList =
                     (List<Span>)sourceTuple.getField(schemaIndex).getValue();
-
+            Collections.sort(spanList, tokenOffsetComp);
```

Comment:
sort is required for combining the span info for the phrase search. Suppose target query is "new is york" , and document contains multiple "new" and "york" term usage, then lucene will return all "new " occurences followed by all "york" occurences, thus to combine span we require to sort it by termOffset